src/fromager/commands/lint_requirements.py

```diff
@@ -0,0 +1,52 @@
+import fileinput
+import glob
+import itertools
+import logging
+import sys
+
+import click
+from packaging.requirements import InvalidRequirement, Requirement
+
+logger = logging.getLogger(__name__)
+
+
+@click.command()
+@click.argument("constraints_wildcard", type=click.Path(exists=False))
+@click.argument("requirements_wildcard", type=click.Path(exists=False))
```

Comment:
If we're going to just pass a bunch of files to the command, we don't need to know whether they are constraints files or requirements files. The format is the same, and we don't need to apply different rules. So this argument could just be a single input that takes [multiple values](https://click.palletsprojects.com/en/stable/arguments/#multiple-arguments). And then we would not need to expand the wildcard ourselves, we can let the shell do that, and we only need a single `for` loop.

Reply:
Constraints file and requirements files are not the same. Constraints files cannot contain optional dependencies (e.g. `package[extra]`).

Reply:
OK, that's a fair point. At the same time, we do not currently test for that downstream and we're mostly trying to move the feature we have downstream upstream as-is. Let's do that, then @rd4398 can add another ticket here to improve the linter to check that extra rule.

Reply:
Okay, makes sense. I have pushed the latest code with changes

Also, I have created the issue here https://github.com/python-wheel-build/fromager/issues/546